runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/WindowingStrategyTranslation.java

```diff
@@ -173,19 +173,61 @@ public static TimestampCombiner timestampCombinerFromProto(RunnerApi.OutputTime
    */
   public static SdkFunctionSpec toProto(
       WindowFn<?, ?> windowFn, @SuppressWarnings("unused") SdkComponents components) {
-    return SdkFunctionSpec.newBuilder()
-        // TODO: Set environment ID
-        .setSpec(
-            FunctionSpec.newBuilder()
-                .setUrn(SERIALIZED_JAVA_WINDOWFN_URN)
-                .setParameter(
-                    Any.pack(
-                        BytesValue.newBuilder()
-                            .setValue(
-                                ByteString.copyFrom(
-                                    SerializableUtils.serializeToByteArray(windowFn)))
-                            .build())))
-        .build();
+    // TODO: Set environment IDs
+    if (windowFn instanceof GlobalWindows) {
```

Comment:
These should probably translated similarly to how we're translating the PTransform Payloads (i.e. with a registrar and some interface that can translate to and from a known type in some meaningful format. We probably want to wait until we've determined that API, though.)